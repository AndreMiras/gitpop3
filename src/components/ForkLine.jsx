import React from 'react';
import PropTypes from 'prop-types';
import timeSince from '../utils/time';
import RepoLink from './RepoLink';

const ForkLine = ({ info }) => (
  <tr>
    <td><RepoLink nameWithOwner={info.nameWithOwner} /></td>
    <td>{info.stargazerCount}</td>
    <td>{info.forkCount}</td>
    <td>{info.object.history.totalCount}</td>
    <td>{timeSince(new Date(info.object.committedDate))}</td>
  </tr>
);
ForkLine.propTypes = {
  info: PropTypes.shape({
    nameWithOwner: PropTypes.string.isRequired,
    stargazerCount: PropTypes.number.isRequired,
    forkCount: PropTypes.number.isRequired,
    object: PropTypes.shape({
      committedDate: PropTypes.string.isRequired,
      history: PropTypes.shape({
        totalCount: PropTypes.number.isRequired,
      }).isRequired,
    }).isRequired,
  }).isRequired,
};

export default ForkLine;
