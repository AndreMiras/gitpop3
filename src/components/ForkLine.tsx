import { FunctionComponent } from "react";
import { Node } from "../utils/types";
import timeSince from "../utils/time";
import RepoLink from "./RepoLink";

type ForkLineProps = {
  info: Node;
};

const ForkLine: FunctionComponent<ForkLineProps> = ({ info }) => (
  <tr>
    <td>
      <RepoLink nameWithOwner={info.nameWithOwner} />
    </td>
    <td>{info.stargazerCount}</td>
    <td>{info.forkCount}</td>
    <td>{info.object.history.totalCount}</td>
    <td>{timeSince(new Date(info.object.committedDate))}</td>
  </tr>
);

export default ForkLine;
