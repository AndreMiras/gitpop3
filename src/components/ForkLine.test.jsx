import React from 'react';
import renderer from 'react-test-renderer';
import ForkLine from './ForkLine';

test('renders', () => {
  const info = {
    nameWithOwner: 'django/django',
    stargazerCount: 123,
    forkCount: 321,
    object: {
      committedDate: '2020-08-29T14:23:26Z',
      history: {
        totalCount: 1234,
      },
    },
  };
  const tree = renderer.create(
    <ForkLine info={info} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
