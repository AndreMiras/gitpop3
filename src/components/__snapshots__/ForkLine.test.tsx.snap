// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<tr>
  <td>
    <a
      href="https://github.com/django/django"
    >
      django/django
    </a>
  </td>
  <td>
    123
  </td>
  <td>
    321
  </td>
  <td>
    1234
  </td>
  <td>
    3 months ago
  </td>
</tr>
`;
